src/herder/TransactionQueue.cpp

```diff
@@ -337,11 +384,13 @@ TransactionQueue::canAdd(TransactionFrameBasePtr tx,
         ban({tx});
         if (canAddRes.second != 0)
         {
-            tx->getResult().result.code(txINSUFFICIENT_FEE);
-            tx->getResult().feeCharged = canAddRes.second;
-            return TransactionQueue::AddResult::ADD_STATUS_ERROR;
+            AddResult result(TransactionQueue::AddResultCode::ADD_STATUS_ERROR,
+                             tx, txINSUFFICIENT_FEE);
+            result.txResult.value()->getResult().feeCharged = canAddRes.second;
+            return result;
         }
-        return TransactionQueue::AddResult::ADD_STATUS_TRY_AGAIN_LATER;
+        return {TransactionQueue::AddResultCode::ADD_STATUS_TRY_AGAIN_LATER,
```

Comment:
nit: This (and line 585) are inconsistent with the majority of return statements here where we call the constructor explicitly.